addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1345,6 +1345,11 @@ var SnippetsMenu = Widget.extend({
         defs.push((async () => {
             await this._loadSnippetsTemplates();
             await this._updateInvisibleDOM();
+            this.snippetEditors.forEach(editor => {
+                if (!editor.$target[0].dataset.visibility) {
```

Comment:
`dataset.visibility` seems to be linked to the `ConditionalVisibility` option which has nothing to do with the cookie bar AFAIK. Can you explain (here on this PR) how this can be the right fix?

For me the fix should probably not be on the edit mode side. When you leave edit mode, each snippet is saved with their default visibility (e.g. modal are closed). If a modal is opened outside of edit mode (like the cookie bar), it should be closed before entering edit mode... thanks to the code who opened it (probably the related public widget? (s_popup/000.js)). (Although I thought it was already the case. If the issue is not that... then I don't understand what this commit does).

Reply:
>  it should be closed before entering edit mode... thanks to the code who opened it (probably the related public widget? (s_popup/000.js)). (Although I thought it was already the case

That's not the issue, things are closed as they should. The issue here is that, as I mentioned here https://github.com/odoo/odoo/pull/100633#discussion_r987756318 (didn't get any answer to the questions btw), contenteditable receives a height, despite being "empty" (its content is hidden). The same issue can be reproduced with the regular popup if you play a bit with it, despite being closed, by just the fact it is in the DOM. Since the snippet `<div>` is never hidden but only its child is, there is the bug.

Note that when toggling the cookies bar visibility in edit mode (showing it and hiding it again), that white bar is not displayed anymore. That info might be useful for your investigation.

Also, please check the code I just found about that: https://github.com/odoo/odoo/blame/16.0/addons/website/static/src/js/editor/snippets.options.js#L2652-L2661

It seems to already be code for this but clearly it's not doing its job anymore. Maybe during a refactor `onTargetHide` is not called as it was at the time they made that. That would explain why we already fixed that issue but it came back.

Reply:
@rdeodoo Now I am a little bit confused, you want me to do a generic fix or only for the cookie bar for now because the link(https://github.com/odoo/odoo/blame/16.0/addons/website/static/src/js/editor/snippets.options.js#L2652-L2661) is only for a cookie bar.

Reply:
Hello @ibo-odoo 

What is confusing? The fact that the code I found of the editor team was doing a specific fix for cookies bar? I'd still do a generic fix for all `s_popup`, not only cookies_bar.
I don't believe it change anything about what QSM and I said about where the fix should be put, the use of `visibility` attribute and figuring the history of the code I found to see why it's not working anymore.

Reply:
For @qsm-odoo 's comment - There is one tour name - `conditional_visibility_1`. In this when we select the 'Text - Image' snippet then we add the `visibility` attribute value "conditionally" and select Email, after saving it when we click on edit it will be there but if I don't put the `visibility` attribute in the condition it will be invisible because without visibility condition it will invisible all 'o_snippet_invisible' snippets. As it applies to all snippets not only cookie_bar.

Reply:
The more reason to adapt like I said and make something specific to popups whose div should be hidden alongside their inner modal :+1: Hiding all o_snippet_invisible snippets when entering edit mode and having some condition to exclude some snippets for which we don't want that auto-hide when entering edit mode is not the way to go.

Reply:
Will be done at https://github.com/odoo/odoo/pull/103382
Let me know if you want me to push that commit in this PR to keep track of what was discussed.